5calls/app/src/androidTest/java/org/a5calls/android/a5calls/controller/MainActivityErrorTest.java

```diff
@@ -0,0 +1,234 @@
+                Thread.sleep(3000); // Increased wait time from 2000 to 3000ms
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+
+            // Set the idling resource to idle
+            idlingResource.setIdle(true);
+
+            // Check that the error message is displayed
+            onView(withText(R.string.request_error))
+                    .check(matches(isDisplayed()));
+
+            // Also verify that the RecyclerView is displayed (but empty)
+            onView(withId(R.id.issues_recycler_view)).check(matches(isDisplayed()));
+
+            // Close the activity
+            scenario.close();
+        } finally {
+            // Unregister the idling resource
+            IdlingRegistry.getInstance().unregister(idlingResource);
+        }
+    }
+
+    @Test
+    public void testJsonErrorDisplaysSnackbar() {
+        // Set up mock to return malformed JSON
+        HttpResponse response = new HttpResponse(200, new ArrayList<>(), "Not valid JSON".getBytes());
+        mHttpStack.setResponseToReturn(response);
+
+        // Create a custom RequestQueue with our mock HTTP stack
+        BasicNetwork basicNetwork = new BasicNetwork(mHttpStack);
+        FakeRequestQueue requestQueue = new FakeRequestQueue(basicNetwork);
+        requestQueue.start();
+
+        // Replace the app's RequestQueue with our mock
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setRequestQueue(requestQueue);
+
+        // Create a new FiveCallsApi with our mock RequestQueue
+        String callerId = AccountManager.Instance.getCallerID(
+                InstrumentationRegistry.getInstrumentation().getTargetContext());
+        FiveCallsApi api = new FiveCallsApi(callerId, requestQueue);
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setFiveCallsApi(api);
+
+        // Create an idling resource to wait for the error to be displayed
+        final ConditionIdlingResource idlingResource = new ConditionIdlingResource();
+        IdlingRegistry.getInstance().register(idlingResource);
+
+        try {
+            // Launch the activity
+            ActivityScenario<MainActivity> scenario = ActivityScenario.launch(MainActivity.class);
+
+            // The activity will automatically make a request for issues
+            // Wait longer for the error to be processed and Snackbar to be displayed
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+
+            // Set the idling resource to idle
+            idlingResource.setIdle(true);
+
+            // Since we observed the same Snackbar appearing for both tests,
+            // let's check for the request_error message instead
+            onView(withText(R.string.request_error))
```

Comment:
this comment seems wrong. we don't seem to observe the snack bar in either test, we just look for the error text. also both tests look for R.string.request_error.

suggest either having a `checkErrorDisplayed()` that can be called for both tests (they do the same thing?) or at least fixing this comment.

Reply:
fixed comment - yeah it was trying to refer to the fact that I watched it display in the emulator with my human :eyes: ..poorly phrased in the comment